src/main/java/org/jenkinsci/account/Myself.java

```diff
@@ -120,17 +118,21 @@ public HttpResponse doChangePassword(
             @QueryParameter String newPassword2
     ) throws Exception {
 
-        // verify the current password
-        try {
-            parent.connect(dn,password).close();
-        } catch (javax.naming.AuthenticationException e) {
+        if ( "".equals(password))
```

Comment:
Doesn't `fixEmpty` does essentially the same thing?

Reply:
Indeed it does the same same things but I don't understand the benefit to create and use a function for that as "".equals(myString) return false if 'myString' is null

Reply:
`fixEmpty(var)` returns null both if it's null and if it's "empty".

Yours distinguishes between the two.

Reply:
https://docs.oracle.com/javase/tutorial/java/data/comparestrings.html
As far as I understand ```"".equals(password)``` return True if password is a string with no character, and false if password is a different object than a string or different string